crates/gateway-types/src/reply.rs

```diff
@@ -739,46 +720,3 @@ pub mod add_transaction {
         }
     }
 }
-
-#[cfg(test)]
-mod tests {
-    /// The aim of these tests is to make sure pathfinder is still able to correctly
-    /// deserialize replies from the mainnet sequencer when it still is using some
-    /// previous version of cairo while at the same time the goerli sequencer is
-    /// already using a newer version.
-    mod backward_compatibility {
-        use super::super::{StateUpdate, Transaction};
-        use starknet_gateway_test_fixtures::*;
-
-        #[test]
-        fn block() {
-            use super::super::MaybePendingBlock;
-
-            serde_json::from_str::<MaybePendingBlock>(v0_8_2::block::GENESIS).unwrap();
-            serde_json::from_str::<MaybePendingBlock>(v0_8_2::block::NUMBER_1716).unwrap();
-            serde_json::from_str::<MaybePendingBlock>(v0_8_2::block::PENDING).unwrap();
-            // This is from integration starknet_version 0.10 and contains the new version 1 invoke transaction.
-            serde_json::from_str::<MaybePendingBlock>(integration::block::NUMBER_216591).unwrap();
-            // This is from integration starknet_version 0.10.0 and contains the new L1 handler transaction.
-            serde_json::from_str::<MaybePendingBlock>(integration::block::NUMBER_216171).unwrap();
-            // This is from integration starknet_version 0.10.1 and contains the new deploy account transaction.
-            serde_json::from_str::<MaybePendingBlock>(integration::block::NUMBER_228457).unwrap();
-        }
-
-        #[test]
-        fn state_update() {
-            // FIXME(0.10): update these fixtures once 0.10 is on mainnet
-
-            // These fixtures do not contain nonces property (0.10 owards).
-            serde_json::from_str::<StateUpdate>(v0_9_1::state_update::GENESIS).unwrap();
-            serde_json::from_str::<StateUpdate>(v0_9_1::state_update::PENDING).unwrap();
-            // This is from integration starknet_version 0.10 and contains the new nonces field.
-            serde_json::from_str::<StateUpdate>(integration::state_update::NUMBER_216572).unwrap();
-        }
-
-        #[test]
-        fn transaction() {
-            serde_json::from_str::<Transaction>(v0_8_2::transaction::INVOKE).unwrap();
-        }
-    }
-}
```

Comment:
I'm not sure if this is safe to remove? I'm worried about transaction's mainly I guess since they're stored as is?

Reply:
I don't think we should remove this. The comment for the `test` module is slightly misleading I think because these types are used to serialize data into the SQLite, too, and we can't break compatibility there.

I think the changes do defaulted attributes are fine though: we're parsing the JSON response from the feeder gateway into these types, and then re-serialize those values into the database. This means that we can remove serde(default) attributes but can't change much else without breaking compatibility with our database.

Reply:
From what I can tell, only the transactions are kept in storage as is - state updates are mapped to be inline with rpc types, and block headers are destructured anyway.

So I should be able to remove everything except the transaction one?

Reply:
Yeah, you're right -- it should be only transactions.